crates/sui-indexer/src/apis/read_api.rs

```diff
@@ -173,13 +182,27 @@ impl ReadApiServer for ReadApi {
     }
 
     async fn get_latest_checkpoint_sequence_number(&self) -> RpcResult<CheckpointSequenceNumber> {
+        if self
+            .method_to_be_forwarded
+            .contains(&"get_latest_checkpoint_sequence_number".to_string())
+        {
+            return self.fullnode.get_latest_checkpoint_sequence_number().await;
+        }
         Ok(self.get_latest_checkpoint_sequence_number().await? as u64)
     }
 
     async fn get_checkpoint(&self, id: CheckpointId) -> RpcResult<Checkpoint> {
-        self.fullnode.get_checkpoint(id).await
+        if self
+            .method_to_be_forwarded
+            .contains(&"get_checkpoint".to_string())
+        {
+            return self.fullnode.get_checkpoint(id).await;
+        }
+        Ok(self.get_checkpoint(id).await?)
```

Comment:
Are we still planning to compare the responses for integrity check?

Reply:
yes, the comparison can happen on a "client" side, for example a read script hit both FN and indexer and compare results there.